Packs/Base/Scripts/DBotMLFetchData/DBotMLFetchData.py

```diff
@@ -626,22 +631,60 @@ def extract_data_from_incidents(incidents, input_label_field=None):
                   'rank': '#{}'.format(i + 1)})
-    load_external_resources()
-    X, exceptions_log, exception_indices, timeout_indices, durations\
-        = extract_features_from_all_incidents(incidents_df, label_fields)
+    custom_fields = [col for col in incidents_df.columns if col not in LABEL_FIELDS_BLACKLIST]
+    subject_field_exists = EMAIL_SUBJECT_FIELD in incidents_df.columns
+    body_field_exists = EMAIL_BODY_FIELD in incidents_df.columns
+    html_field_exists = EMAIL_HTML_FIELD in incidents_df.columns
+    headers_field_exists = EMAIL_HEADERS_FIELD in incidents_df.columns
+    attachments_field_exists = EMAIL_ATTACHMENT_FIELD in incidents_df.columns
+    n_missing_subject_field = sum(incidents_df[EMAIL_SUBJECT_FIELD].isnull()) if subject_field_exists else n_incidents
+    n_missing_body_field = sum(incidents_df[EMAIL_BODY_FIELD].isnull()) if body_field_exists else n_incidents
+    n_missing_html_field = sum(incidents_df[EMAIL_HTML_FIELD].isnull()) if html_field_exists else n_incidents
+    n_missing_headers_field = sum(incidents_df[EMAIL_HEADERS_FIELD].isnull()) if headers_field_exists else n_incidents
+    n_missing_attachments_field = sum(incidents_df[EMAIL_ATTACHMENT_FIELD].isnull()) \
+        if attachments_field_exists else n_incidents
+    if not (subject_field_exists or body_field_exists or html_field_exists):
+        X = []
+        exceptions_log = []
+        short_text_indices = []
+        exception_indices = []
+        timeout_indices = []
+        durations = []
+    else:
+        load_external_resources()
+        X, exceptions_log, short_text_indices, exception_indices, timeout_indices, durations \
+            = extract_features_from_all_incidents(incidents_df, label_fields)
+
     return {'X': X,
             'n_fetched_incidents': len(X),
             'y': y,
             'log':
                 {'exceptions': exceptions_log,
                  'n_timout': len(timeout_indices),
                  'n_other_exceptions': len(exception_indices),
-                 'durations': durations}
+                 'durations': durations,
+                 'n_fetched_incidents': n_fetched_incidents,
+                 'n_labeled_incidents': n_incidents,
+                 'subject_field_exists': subject_field_exists,
+                 'body_field_exists': body_field_exists,
+                 'html_field_exists': html_field_exists,
+                 'headers_field_exists': headers_field_exists,
+                 'attachments_field_exists': attachments_field_exists,
+                 'n_missing_subject_field': n_missing_subject_field,
+                 'n_missing_body_field': n_missing_body_field,
+                 'n_missing_html_field': n_missing_html_field,
+                 'n_missing_headers_field': n_missing_headers_field,
+                 'n_missing_attachments_field': n_missing_attachments_field,
+                 'n_short_text_fields': len(short_text_indices),
+                 'custom_fields': custom_fields
```

Comment:
I think it will be nice to bring how many unique values each field has.. it will help us to understand the meaning of the field maybe